eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentLifecycle.scala

```diff
@@ -61,6 +62,7 @@ class PaymentLifecycle(sourceNodeId: PublicKey, router: ActorRef, register: Acto
         stop(FSM.Normal)
       } else {
         register ! Register.ForwardShortId(firstHop.lastUpdate.shortChannelId, cmd)
+        if (hops.size > 1) nodeParams.pendingPaymentDb.add(c.paymentHash, firstHop.nextNodeId, c.targetNodeId, firstHop.lastUpdate.cltvExpiryDelta, currentHeight, currentHeight, cmd.expiry)
```

Comment:
This table has slightly confused me. It never seems to delete payments when they have been paid or failed? I had assumed it only had payments that are in flight?

Reply:
Yes, they are never deleted. This table's function is to fixate a delays each payment takes and later use that info for risk analysis when evaluating if future payments to a given payee should be issued or not.

Reply:
Currently pending payments can be extracted from `channel` output, they don't need to be stored in a special table.

Reply:
Yes, you can easily identify payments by channel id+ htlc id, so if you can make `sendpayment` return channel id and htlc id you can then have a `checkpayment` query that is not expensive and does not require too many internal changes.

Reply:
> you can easily identify payments by channel id+ htlc id

payment may hang in one channel for some time and then be re-sent through another channel, as far as I understand the only thing we can surely rely on is payment hash, and all channels should be queried.